pep-9999.rst

```diff
@@ -0,0 +1,109 @@
+proposed deprecating and eventually removing upload support for a handful
+of un(der)used file extensions and distribution types.
+
+In particular, :pep:`527` proposed the removal of the ``bdist_dumb``,
+``bdist_rpm``, ``bdist_dmg``, ``bdist_msi``, and ``bdist_wininst`` distribution
+types, as well as the ``.tar``, ``.tar.bz2``, ``.tar.xz``, ``.tar.Z``,
+``.tgz``, and ``.tbz`` file extensions on distribution filenames.
+
+:pep:`527` was fully enacted with
+`PR #7529 <https://github.com/pypi/warehouse/pull/7529>`_ to Warehouse,
+which was merged on 13 April 2020.
+
+The ``bdist_egg`` format
+------------------------
+
+The ``bdist_egg`` filetype identifies distributions in the
+`Egg format <https://packaging.python.org/en/latest/glossary/#term-Egg>`_. The
+Egg format was introduced by setuptools in 2004 and is roughly equivalent
+in functionality to the
+`Wheel format <https://packaging.python.org/en/latest/glossary/#term-Wheel>`_
+introduced and standardized by :pep:`427` in 2012.
+
+Despite its longevity, the Egg format has had
+`limited adoption on PyPI <https://github.com/pypi/warehouse/issues/10653>`_.
+Some observations from that issue:
+
+* As of December 2021, ``bdist_egg`` uploads account for less than 1% of all
+  built distribution uploads;
+* ``pip`` deprecated its ``--egg`` option in 2016;
+* ``setuptools`` has considered Egg support deprecated since 2019;
+* ``build`` only supports the ``sdist`` and ``bdist_wheel`` filetypes.
+
+Given the above, this PEP proposes the removal of the ``bdist_egg`` format
+under the same justifications presented in :pep:`527`, namely:
+
+1. Egg distributions are of of limited use to the broader ecosystem and
+   therefore represent a non-reciprocal maintenance burden;
+2. Continued support for Egg distributions represents a bandwidth and storage
+   cost to PyPI, as well as to any external mirrors of PyPI.
+
+This PEP *additionally* offers an argument for removal rooted in
+standardization and duplication: the Egg format is not standardized by any
+PEP or other community standard, and overlaps heavily with its standardized
+and well-supported alternative (Wheel).
+
+The ``.egg`` file extension
+---------------------------
+
+The ``.egg`` file extension is used exclusively for distributions of the
+``bdist_egg`` format. As such, it serves no purpose in a scenario where
+PyPI disables new distribution uploads for ``bdist_egg`` distributions.
+
+Removal Process
+===============
+
+This PEP proposes a removal process based on :pep:`527`, but with the following
+modifications:
+
+* During the deprecation period, *all* packages will
+  `continue to be allowed <https://www.youtube.com/watch?v=XNyUALnj8V0>`_
+  to upload ``bdist_egg`` distributions, not just those that have previously
+  uploaded ``bdist_egg`` distributions.
+* During the deprecation period, uploading a ``bdist_egg`` distribution to PyPI
+  will trigger an email to all maintainers of the containing package, rather
```

Comment:
```suggestion
  will trigger an email to all maintainers of the project, rather
```

Isn't it really a "project", not a "package" here? And "containing" seems rather confusing and unnecessary here, IMO.